crates/tests/core/tests/hstrings.rs

```diff
@@ -153,3 +152,43 @@ fn hstring_osstring_equality_combinations() {
     assert_eq!(ss, h);
     assert_eq!(ss, &h);
 }
+
+#[test]
+fn hstring_compat() -> Result<()> {
+    unsafe {
+        use windows::Win32::System::WinRT::*;
+        let hey = HSTRING::from("Hey");
+        let world = HSTRING::from("World");
+        assert_eq!(WindowsCompareStringOrdinal(&hey, &world)?, -1);
+        assert_eq!(WindowsCompareStringOrdinal("Hey", "World")?, -1);
+
+        assert_eq!(WindowsConcatString(&hey, &world)?, "HeyWorld");
+        assert_eq!(WindowsConcatString("Hey", "World")?, "HeyWorld");
+
+        assert_eq!(WindowsCreateString(hey.as_wide())?, "Hey");
+
+        assert_eq!(WindowsDuplicateString(&hey)?, "Hey");
+        assert_eq!(WindowsDuplicateString("Hey")?, "Hey");
+
+        assert_eq!(WindowsGetStringLen(&hey), 3);
+        assert_eq!(WindowsGetStringLen("Hey"), 3);
+        assert_eq!(WindowsGetStringLen(&world), 5);
+        assert_eq!(WindowsGetStringLen("World"), 5);
+
+        assert_eq!(WindowsIsStringEmpty(HSTRING::new()), true);
+        assert_eq!(WindowsIsStringEmpty(""), true);
+        assert_eq!(WindowsIsStringEmpty(&world), false);
+        assert_eq!(WindowsIsStringEmpty("World"), false);
+
+        let mut len = 0;
+        let buffer = WindowsGetStringRawBuffer(&world, &mut len);
+        assert_eq!(std::slice::from_raw_parts(buffer.0, len as _), world.as_wide());
+
+        // We need to drop to windows-sys to call the raw WindowsDeleteString function to avoid double-freeing the HSTRING,
+        // but this test is important as it ensures that the allocators match.
+        let hresult = windows_sys::Win32::System::WinRT::WindowsDeleteString(std::mem::transmute_copy(&*std::mem::ManuallyDrop::new(hey)));
```

Comment:
Somewhat unrelated but should `WindowsDeleteString` take a `ManuallyDrop<HSTRING>` instead of a `HSTRING`?

Reply:
Yes, there's just not metadata for that right now. We could just exclude it from the `windows` crate as there's no correct way to call it and there's no need to call it either.

Reply:
Yep, there's friction for sure.